sdk/src/types/block/output/feature/metadata.rs

```diff
@@ -1,24 +1,56 @@
 // Copyright 2021-2022 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-use alloc::{boxed::Box, string::String, vec::Vec};
-use core::{ops::RangeInclusive, str::FromStr};
-
-use packable::{bounded::BoundedU16, prefix::BoxedSlicePrefix};
+use alloc::{
+    collections::BTreeMap,
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::ops::{Deref, RangeInclusive};
+
+use packable::{
+    bounded::{BoundedU16, BoundedU8},
+    prefix::{BTreeMapPrefix, BoxedSlicePrefix},
+};
 
 use crate::types::block::{output::StorageScore, protocol::WorkScore, Error};
 
 pub(crate) type MetadataFeatureLength =
     BoundedU16<{ *MetadataFeature::LENGTH_RANGE.start() }, { *MetadataFeature::LENGTH_RANGE.end() }>;
 
+pub(crate) type MetadataFeatureKeyLength = BoundedU8<1, { u8::MAX }>;
+pub(crate) type MetadataFeatureValueLength = BoundedU16<0, { u16::MAX }>;
```

Comment:
Shouldn't this use LENGTH_RANGE ?

Reply:
Or is LENGTH_RANGE for the complete serialised thing? Not specific limits for key/value in the TIP?

Reply:
> Not specific limits for key/value in the TIP?

No, just the overall size is limited. We can also limit values by that (or even less because of the required  type, entries count and key bytes), but doesn't change much?